cpp/src/groupby/sort/group_nunique.cu

```diff
@@ -58,57 +62,16 @@ struct is_unique_iterator_fn {
 
   __device__ size_type operator()(size_type i)
   {
-    bool is_input_countable =
+    auto const is_input_countable =
       !nulls || (null_handling == null_policy::INCLUDE || v.is_valid_nocheck(i));
-    bool is_unique = is_input_countable &&
-                     (group_offsets[group_labels[i]] == i ||          // first element or
-                      (not equal.template operator()<T>(i, i - 1)));  // new unique value in sorted
+    auto const is_unique =
+      is_input_countable && (group_offsets[group_labels[i]] == i ||  // first element or
+                             (not equal(i, i - 1)));                 // new unique value in sorted
```

Comment:
Will this cause any out-of-bound issue with `i - 1`?

Reply:
Also should this `operator()` const?

Reply:
> Will this cause any out-of-bound issue with i - 1?

When `i == 0`, the operation will return at `group_offsets[group_labels[i]] == i` thus `i - 1` won't be performed.